freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/FreeIpaClientFactory.java

```diff
@@ -70,42 +73,48 @@ private String toClusterProxyBasepath(String freeIpaClusterCrn) {
         return String.format("%s%s", clusterProxyService.getProxyPath(freeIpaClusterCrn), FreeIpaClientBuilder.DEFAULT_BASE_PATH);
     }
 
-    public FreeIpaClient getFreeIpaClientForStack(Stack stack) throws FreeIpaClientException {
-        LOGGER.debug("Creating FreeIpaClient for stack {}", stack.getResourceCrn());
+    private FreeIpaClient getFreeIpaClient(Stack stack, boolean withPing) throws FreeIpaClientException {
+        stack = stackService.getByIdWithListsInTransaction(stack.getId());
         Status stackStatus = stack.getStackStatus().getStatus();
         if (!stackStatus.isFreeIpaUnreachableStatus()) {
             try {
                 if (clusterProxyService.isCreateConfigForClusterProxy(stack)) {
-                    return getFreeIpaClientBuilderForClusterProxy(stack).build();
+                    return getFreeIpaClientBuilderForClusterProxy(stack).build(withPing);
                 } else {
-                    return getFreeIpaClientBuilder(stack).build();
+                    Exception lastException = new FreeIpaClientException("No instances exist in the stack.");
+                    for (InstanceMetaData instanceMetaData: stack.getInstanceGroups().stream().flatMap(
```

Comment:
please move out the stream into a variable and/or to a method

Reply:
moved